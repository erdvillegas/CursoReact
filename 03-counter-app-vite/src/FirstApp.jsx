import PropTypes from 'prop-types';


const getResult = () => 4 + 4;

const newMesagge = {
  message: 'hola Mundo',
  nombre: 'Erik'
};


export const FirstApp = ( { title, subtitle, name } ) => {

  return (
    <>
      <h1 data-testid = "test-title"> { title }</h1>
      <p>{ subtitle }</p>
      <p>{ name }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

FirstApp.defaultProps = {
  name: 'Erik Villegas',
  subtitle: 'no hay subtitulo u'
}