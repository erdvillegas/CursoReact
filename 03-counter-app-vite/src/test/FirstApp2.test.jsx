/** * @jest-environment jsdom */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../FirstApp';


describe('Pruebas en <FirstApp/>', () => {


    const title = 'Hola, soy el titulo!';
    const subTitle = 'Hola, soy el subTitle!';

    test('debe de hacer match con el snappshot', () => {

        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar el mensaje \'Hola, soy el titulo!\'', () => {

        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(title);
    });
    
    test('debe de mostrar el subtitulo envido por las props', () => {
        
        render(
            <FirstApp 
                title={ subTitle } 
                subtitle={ subTitle }
            />
        );
        
        expect(screen.getAllByText(subTitle).length).toBe(2);
        
    });
    

});