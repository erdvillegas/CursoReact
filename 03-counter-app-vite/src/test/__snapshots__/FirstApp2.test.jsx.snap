// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snappshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy el titulo!
  </h1>
  <p>
    no hay subtitulo u
  </p>
  <p>
    Erik Villegas
  </p>
</div>
`;
